src/fusion_power_plant.cc

```diff
@@ -10,14 +10,98 @@ std::string FusionPowerPlant::str() {
   return Facility::str();
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    //Set up a schedule fill policy with Katie's active/dormant policy code
+    //(not shown here). Do not start the policy yet.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (can_operate) {
```

Comment:
instead of `can_operate` being a Boolean variable, it could be a function that is evaluated each time

Reply:
Agreed, though, if I remember correctly there was some issue with simulation-ending errors which I was catching with the try/catch block (I see you have a comment about that later on as well). This is definitely something to try in the full implementation, just mentioning it in case it doesn't fully work.